applets/clock/clock.c

```diff
@@ -2682,7 +2682,7 @@ run_prefs_edit_save (GtkButton *button, ClockData *cd)
                  */
                 clock_location_is_current (loc);
 
-                cd->locations = g_slist_append (cd->locations, loc);
+                cd->locations = g_slist_append (cd->locations, g_object_ref (loc));
```

Comment:
I don't think it fixes any errors, as it always creates a new reference to a newly created ClockLocation object, fact that leads to a memory leak.

Reply:
First, the fact is that adding a city will lead to segment errors, which have disappeared through this PR segment

Second, the reason for the segment error is that ```g_object_unret```
Here the memory has been freed
```
static void
free_locations (ClockData *cd)
{
        if (cd->locations != NULL) {
                GSList *l;
                for (l = cd->locations; l; l = l->next)
                        g_object_unref (l->data);
                g_slist_free (cd->locations);
        }
        cd->locations = NULL;
}
```

Reply:
It not NULL here.

![VirtualBox_9 Fedora 33_02_03_2021_13_02_30](https://user-images.githubusercontent.com/10171411/109645856-b5f47480-7b57-11eb-8dc4-8ac3dc602ba0.png)

Reply:
You can try adding cities many times, and it does solve my problem，We can wait for anyone with a segment error to test it

Reply:
I think your problem comes from somewhere else.